### PR TITLE
Update docs to reflect current best practice

### PR DESCRIPTION
**Pull request by mkjpryor**

### CODE REVIEW

File: docs/best-practice.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+A
+[sample GitLab CI/CD configuration](https://github.com/stackhpc/azimuth-config/tree/main/.gitlab-ci.yml.sample)
+is provided that implements this workflow for GitLab-hosted repositories.
+
+## Configuration
+
+You are now ready to begin adding configuration to your environments. When building an environment
+for the first time, it is recommended to follow each documentation page in order, beginning with
+the [Deployment method](http://localhost:8000/configuration/02-deployment-method/).
```
Comment (m-bull) [inline docs/best-practice.md:110]:
```suggestion
the [Deployment method](./configuration/02-deployment-method.md).
```

File: docs/configuration/01-prerequisites.md [hunk 2]
```diff
@@ -4,38 +4,107 @@ In order to deploy Azimuth, a small number of prerequisites must be fulfilled.
...
+  * Image (Glance)
+  * Compute (Nova)
+  * Block storage (Cinder)
+  * Network (Neutron)
+    * OVN and OVS/ML2 drivers supported
+  * Load balancer (Octavia)
+    * Amphora and OVN drivers supported
+
+Azimuth has no specific requirement on the version of OpenStack. It is known to work on
+Train and newer.
```
Comment (m-bull) [inline docs/configuration/01-prerequisites.md:19]:
```suggestion
on Train release and newer.
```

File: docs/configuration/01-prerequisites.md [hunk 3]
```diff
@@ -4,38 +4,107 @@ In order to deploy Azimuth, a small number of prerequisites must be fulfilled.
...
+### Networking
+
+OpenStack projects that are used for Azimuth deployments (infra projects) or into which
+Azimuth will deploy platforms on behalf of users (workload projects) require access to a
+network that is shared as `external` in Neutron, on which floating IPs can be allocated.
+
+This network must provide egress to the internet, but does not need to provide ingress
+from the internet. This network should be given the Neutron tag `portal-external` so that
+it can be correctly detected by Azimuth, especially in the case where there are multiple
+external networks.
```
Comment (m-bull) [inline docs/configuration/01-prerequisites.md:30]:
```bash
openstack network set --tag portal-external ${external_network_name}
```

File: docs/configuration/03-kubernetes-config.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+To configure Kubernetes clusters to use volume-backed instances (i.e. use a Cinder
+volume as the root disk), the following variables can be used:
+
+```yaml  title="environments/my-site/inventory/group_vars/all/variables.yml"
+#### For the HA cluster ####
+
+# The size of the root volumes for Kubernetes nodes
+capi_cluster_root_volume_size: 100
+# The volume type to use for root volumes for Kubernetes nodes
+capi_cluster_root_volume_type: nvme
```
Comment (m-bull) [inline docs/configuration/03-kubernetes-config.md:104]:
```bash
openstack volume type list
```

File: docs/configuration/03-kubernetes-config.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+#### For tenant clusters ####
+
+azimuth_capi_operator_capi_helm_root_volume_size: 100
+azimuth_capi_operator_capi_helm_root_volume_type: nvme
+```
+
+## Load-balancer provider
+
+If the target cloud uses [OVN networking](https://www.ovn.org/en/), and the
+[OVN Octavia driver](https://docs.openstack.org/ovn-octavia-provider/latest/admin/driver.html)
```
Comment (m-bull) [inline docs/configuration/03-kubernetes-config.md:115]:
```suggestion
[OVN Octavia provider](https://docs.openstack.org/ovn-octavia-provider/latest/admin/driver.html)
```

File: docs/configuration/03-kubernetes-config.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+is enabled, then Kubernetes clusters should be configured to use the OVN provider for
+any load-balancers that are created:
+
+```yaml  title="environments/my-site/inventory/group_vars/all/variables.yml"
+#### For the HA cluster ####
+
+# The provider for the API server load-balancer created by Cluster API
+capi_cluster_apiserver_loadbalancer_provider: ovn
```
Comment (m-bull) [inline docs/configuration/03-kubernetes-config.md:123]:
```bash
openstack loadbalancer provider list
```

File: docs/configuration/10-kubernetes-clusters.md [hunk 7]
```diff
@@ -173,13 +117,27 @@ azimuth_capi_operator_cluster_templates_extra:
...
 [proxy cache](https://goharbor.io/docs/2.1.0/administration/configure-proxy-cache/)
-to avoid pulling images from the internet. The Harbor registry will be made available at
-`registry.<ingress base domain>`, e.g. `registry.apps.example.org`.
+to limit the number of times that images that are pulled directly from the internet.
+
+If enabled, the Harbor registry will be made available at `registry.<ingress base domain>`,
+e.g. `registry.azimuth.example.org` and will be configured for tenant clusters by default.
+
+By default, a single proxy cache is defined for [Docker Hub](https://hub.docker.com/) in order
+to mitigate the [rate limiting](https://docs.docker.com/docker-hub/download-rate-limit/).
```
Comment (m-bull) [inline docs/configuration/10-kubernetes-clusters.md:128]:
```suggestion
to mitigate [rate limiting](https://docs.docker.com/docker-hub/download-rate-limit/).
```

File: docs/configuration/10-kubernetes-clusters.md [hunk 8]
```diff
@@ -203,19 +161,12 @@ The Harbor registry can be disabled entirely:
 
 ```yaml  title="environments/my-site/inventory/group_vars/all/variables.yml"
 harbor_enabled: no
-```
+```
```
Comment (m-bull) [inline docs/configuration/10-kubernetes-clusters.md:164]:
```suggestion
```
```

**Review by m-bull — CHANGES_REQUESTED** [review]

File: docs/best-practice.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+meets the [prerequisites](./configuration/01-prerequisites.md).
+
+## OpenStack projects
+
+Azimuth is usually deployed on the cloud that is being targetted for workloads. It is
+recommended to have three OpenStack projects for a production Azimuth deployment:
+
+  * One for the highly-available (HA) production deployment, e.g. `azimuth-production`
+  * One for a HA staging deployment, e.g. `azimuth-staging`
+  * One for all-in-one (AIO) deployments for validating changes, e.g. `azimuth-cicd`
```
[the remark this answers is not in the document]
Comment (mkjpryor) [reply docs/best-practice.md:18]: The "an" doesn't really make sense here. Probably just "All-on-one..." I guess

**Review by m-bull — APPROVED** [review]
Nice

**Comment by bbezak** [comment]
nice, btw, I couldn't find git-crypt for rocky9 in repo anywhere (not even in epel9). However binary from github works, but it needs to have compat-openssl11 installed:

`wget https://github.com/AGWA/git-crypt/releases/download/0.7.0/git-crypt-0.7.0-linux-x86_64
./git-crypt-0.7.0-linux-x86_64: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory
sudo dnf install compat-openssl11.x86_64

sudo cp git-crypt-0.7.0-linux-x86_64 /usr/local/bin/git-crypt
sudo chmod 711 /usr/local/bin/git-crypt

git-crypt version
git-crypt 0.7.0`

**Pull request merged**